packages/ifpack2/src/Ifpack2_Container.hpp

```diff
@@ -1069,7 +1069,7 @@ void ContainerImpl<MatrixType, LocalScalarType>::DoGSBlock(
     //Use the KokkosSparse internal matrix for low-overhead values/indices access
     //But, can only do this if the matrix is accessible directly from host, since it's not a DualView
     auto localA = this->inputCrsMatrix_->getLocalMatrix();
-    typedef typename decltype(localA)::size_type size_type;
+    using size_type = typename decltype(localA)::size_type;
```

Comment:
You know the type of `inputCrsMatrix_` at this point (it's a Tpetra::CrsMatrix specialization), so you should have a typedef for it.  Thus, you don't need `decltype`; just use `crs_matrix_type::local_matrix_type` directly.